### PR TITLE
Made a couple of member protected

### PR DESCRIPTION
**Pull request by YoupHulsebos**
I needed this in a derived class of the Auth0Client. Thought it would be nice to have in the original.

### CODE REVIEW

**Comment by dschenkelman** [comment]
Could you share a bit more about your use case? We have noticed that extending Auth0Client sometimes requires this info if subclassing, but making members protected might not be the best/only solution.

**Comment by YoupHulsebos** [comment]
Sure thing. I'm integrating the auth0 account into the android account manager. We want the account to show up there to add that native feeling to the app.

To do so you have to write a an android service that binds to a implementation of AbstractAccountAuthenticator. In this implementation you have to deliver an intent for android to start if the user adds an account from the settings page. This is just a little different from the default implementation that the Auth0Client delivers. 

The SendLoginAsync starts the intent itself and has no knowledge of where the account wil be stored. We wanted an implementation that delivers an intent that is fully aware where it should store the account if succesfull and it should not start itself.

Does that make any sense? If you need more info just let me know.

**Comment by dschenkelman** [comment]
Great, I get the general idea and think the update might be beneficial for other lib consumers to thank you!

Would you mind making a few changes to the PR?

I think it would be nice to:
* Add all constants to a different `public` class. There is no `private` information in them anyways. We could make it `protected` but there is no point in that IMO.
* Change the `private readonly` fields into readonly `protected` properties.

I'll review the `SetupCurrentUser` change when I get a bit more time.

/ccing: @siacomuzzi @sandrinodimattia @ntotten in case they have more feedback.

**Comment by YoupHulsebos** [comment]
Hi Damian,

I processed your feedback. An other change I made with in the new commit is making the setter for CurrentUser public. I needed this to set my current user in the Auth0Client so I can get a new token for the user. 
Let me know if you have any more feedback.

Youp

File: src/Auth0Client/Auth0Client.cs [hunk 1]
```diff
@@ -32,7 +25,7 @@ public Auth0Client (string domain, string clientId)
 			this.DeviceIdProvider = new DeviceIdProvider();
 		}
 
-		public Auth0User CurrentUser { get; private set; }
+		public Auth0User CurrentUser { get; set; }
```
Comment (dschenkelman) [inline src/Auth0Client/Auth0Client.cs]: I think the setter should be `protected`.

File: src/Auth0Client/Auth0Client.cs [hunk 2]
```diff
@@ -303,9 +296,9 @@ private static bool ScopeHasOfflineAccess(string scope)
 				.Any(e => e.Equals("offline_access", StringComparison.InvariantCultureIgnoreCase));
 		}
 
-		private void SetupCurrentUser(IDictionary<string, string> accountProperties)
+		protected void SetupCurrentUser(IDictionary<string, string> accountProperties)
```
Comment (dschenkelman) [inline src/Auth0Client/Auth0Client.cs]: I'm good with this change.

**Comment by dschenkelman** [comment]
Great! Just left a minor comment regarding that last change.

All the constant files were added as project links, right? If that is the case once this is done I'll download the latest version and if everything is working merge this :)

**Comment by YoupHulsebos** [comment]
Jep, the files were added by reference. 
Sounds great! Looking forward to the merge.

**Comment by dschenkelman** [comment]
Merged https://github.com/auth0/Xamarin.Auth0Client/pull/10 as I did added small change.

It might take some time until we publish a new version of the component since the publishing process takes some time, which is why we usually batch changes together.

**Pull request merged**